clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -357,6 +398,74 @@ void SemaHLSL::handleNumThreadsAttr(Decl *D, const ParsedAttr &AL) {
     D->addAttr(NewAttr);
 }
 
+static bool isValidWaveSizeValue(unsigned Value) {
+  return (Value >= 4 && Value <= 128 && ((Value & (Value - 1)) == 0));
```

Comment:
Can we instead use `llvm::isPowerOf2_32`? That is clearer to read.

Reply:
Updated.